### PR TITLE
added processing of meta pages with deleted rows.

### PR DESCRIPTION
**Pull request by Yana-Guseva**

### CODE REVIEW

**Comment by PCaff** [comment]
This is a good start. It does not solve the deleted records issue. But, it does handles reading in the pages.

**Comment by Yana-Guseva** [comment]
Hi @PCaff , did you manage to reproduce the deleted rows issue on any test data? I thought that all_rand_normal_with_deleted.sas7bdat file that you attached earlier (thanks!) contains deleted rows and thus can be used for test this issue. 
I assumed that my fix will only work if the data has been deleted from the page with type 'mix'. That is, I think that you can get an error on other files if the rows have been deleted from a 'data' or 'meta' pages.

**Comment by PCaff** [comment]
@Yana-Guseva So what I've gathered so far is that page type of 640 is very similar to the MIX type, and maybe identical. What I haven't been able to figure out is how to read in the deleted records and then recognize which of the records is deleted and which are not. I'm having trouble finding the pointers for the deleted records. If you run your code on the dataset, you return all records (inclusive of the deleted ones). Please correct me if I'm wrong, those were the results when I ran this fix (though I used my own version of code)

File: src/test/resources/csv/all_rand_normal_with_deleted.csv [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+6,-1.30601106460683,-0.191645716555782,-0.890895281080635,-1.01420554584294,0.239205026896137,-1.36956281497245,0.367116358963696
+7,-0.182529555280925,1.70588149747924,-0.648252455053375,-0.113856279856997,-1.59414475661726,-0.0716625039715658,0.422675801034446
+8,1.55587487228279,-0.240535685327487,-0.987747968432627,0.379810225477961,-0.616028300364221,-1.87098495560613,0.63376130380024
+9,0.747240400228516,0.0760947073326997,-0.755460377522965,0.146127904800772,-1.22177835494399,-0.709271266564636,2.57184269865151
+10,-0.279883956321368,1.47435132507343,-0.948816009788528,-0.844456583968411,0.139119584115725,1.35427048218451,-1.27013451653836
+20,1.41199698228818,0.836223566992313,0.266081420325081,-1.07577894523886,-0.776099337225336,-0.292190865045215,-0.629139226286168
+30,-1.25875867828822,0.185198950004779,-0.116324903204208,-1.03625723019971,-0.0170051390322138,-0.322461408145671,0.00632738879553362
+40,0.891389660634871,-0.883007989589512,-0.654361699302352,1.26265202786226,1.13288576340466,-0.0927398141356724,0.0672848024456528
+50,2.15000039664857,0.749410953295052,-1.16764981806199,0.466677499486481,-0.0552601977313928,0.0143111301094924,-0.39946607189219
```
Comment (PCaff) [inline src/test/resources/csv/all_rand_normal_with_deleted.csv]: @Yana-Guseva So this record in the SAS version was removed. The way SAS works in some cases is that is does not purge from the file in non compressed versions. We need to inspect the Subheaders and find the pointers that say "this record is invalid". I believe it is hidden somewhere in the Row Size Subheader.

Comment (PCaff) [reply src/test/resources/csv/all_rand_normal_with_deleted.csv]: to make it clear, the record with the value '50' in the first column

**Comment by PCaff** [comment]
@Yana-Guseva I will be updating your pull request with new logic in the next few days or so. Stay tuned!

**Comment by PCaff** [comment]
@printsev @Yana-Guseva Could one of you possibly provide write access to me for this repo, please?

**Comment by printsev** [comment]
@PCaff -- I believe you should have access to create a new PR on top of this one, could you please check?

**Comment by PCaff** [comment]
@printsev - Hey again. I don't think I have access to create a new PR ontop of this one.

**Comment by TimurMikhaliov** [comment]
> @printsev @Yana-Guseva Could one of you possibly provide write access to me for this repo, please?

Hi @PCaff  this is done now. Could you please check everything is working for you? 
Thanks

**Comment by PCaff** [comment]
@TimurMikhaliov Great! I'll fiddle around today and make sure I have the proper access levels. Thank you!

**Comment by PCaff** [comment]
@printsev @Yana-Guseva Please have a look at this PR when you get a chance.

**Review by printsev — APPROVED** [review]

**Pull request merged**